yrs/src/observer.rs

```diff
@@ -0,0 +1,274 @@
+use crate::atomic::AtomicRef;
+use std::fmt::{Debug, Formatter};
+use std::sync::atomic::{AtomicU32, Ordering};
+use std::sync::Arc;
+
+pub type SubscriptionId = u32;
+
+#[derive(Debug)]
+pub struct Observer<F: Clone> {
```

Comment:
Observer replaces an old `EventHandler`. It's thread safe without locks and it's using atomic hardware intrinsic ops combined with copy-on-write pattern to perform updates.